### PR TITLE
Add find by id method overload

### LINKED ISSUE #61: Add an overload for `IRepositoryReadConductor#FindById` which accepts additional filters
@ralphjmcintyre and I recently ran into a situation in a client project where, based on a user's role in the system, we either want to include or exclude "unpublished" records. It would be useful, and also potentially more efficient, if we were able to do basically "find by ID, if it also matches this filter".

Could be something like: 

`_readConductor.FindById(id: id, filter: e => e.PublishedOn.HasValue)`

### PR DESCRIPTION
**Pull request by SaidShah**
…the XML docs

- [ x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

Closes #61

### CODE REVIEW

File: src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md [hunk 1]
```diff
@@ -207,17 +207,36 @@
...
+  - [FindById(id,includeProperties)](#M-AndcultureCode-CSharp-Core-Interfaces-Conductors-IRepositoryReadConductor`1-FindById-System-Int64,System-Linq-Expressions-Expression{System-Func{`0,System-Object}}[]- 'AndcultureCode.CSharp.Core.Interfaces.Conductors.IRepositoryReadConductor`1.FindById(System.Int64,System.Linq.Expressions.Expression{System.Func{`0,System.Object}}[])')
+  - [FindById(id,ignoreQueryFilters,includeProperties)](#M-AndcultureCode-CSharp-Core-Interfaces-Conductors-IRepositoryReadConductor`1-FindById-System-Int64,System-Boolean,System-Linq-Expressions-Expression{System-Func{`0,System-Object}}[]- 'AndcultureCode.CSharp.Core.Interfaces.Conductors.IRepositoryReadConductor`1.FindById(System.Int64,System.Boolean,System.Linq.Expressions.Expression{System.Func{`0,System.Object}}[])')
+  - [FindById(id,includeProperties)](#M-AndcultureCode-CSharp-Core-Interfaces-Conductors-IRepositoryReadConductor`1-FindById-System-Int64,System-String[]- 'AndcultureCode.CSharp.Core.Interfaces.Conductors.IRepositoryReadConductor`1.FindById(System.Int64,System.String[])')
```
Comment (jstevenson) [inline src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md:214]: I am not sure if this should be sorted by input names, but you at least group the two `FindById(id,includeProperties)` methods.

File: src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md [hunk 2]
```diff
@@ -207,17 +207,36 @@
...
+  - [FindById(id,includeProperties)](#M-AndcultureCode-CSharp-Core-Interfaces-Data-IRepository`1-FindById-System-Int64,System-Linq-Expressions-Expression{System-Func{`0,System-Object}}[]- 'AndcultureCode.CSharp.Core.Interfaces.Data.IRepository`1.FindById(System.Int64,System.Linq.Expressions.Expression{System.Func{`0,System.Object}}[])')
+  - [FindById(id,ignoreQueryFilters,includeProperties)](#M-AndcultureCode-CSharp-Core-Interfaces-Data-IRepository`1-FindById-System-Int64,System-Nullable{System-Boolean},System-Linq-Expressions-Expression{System-Func{`0,System-Object}}[]- 'AndcultureCode.CSharp.Core.Interfaces.Data.IRepository`1.FindById(System.Int64,System.Nullable{System.Boolean},System.Linq.Expressions.Expression{System.Func{`0,System.Object}}[])')
+  - [FindById(id,includeProperties)](#M-AndcultureCode-CSharp-Core-Interfaces-Data-IRepository`1-FindById-System-Int64,System-String[]- 'AndcultureCode.CSharp.Core.Interfaces.Data.IRepository`1.FindById(System.Int64,System.String[])')
```
Comment (jstevenson) [inline src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md:236]: Same here

File: src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md [hunk 3]
```diff
@@ -2886,6 +2915,96 @@ true
...
+##### Parameters
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| id | [System.Int64](http://msdn.microsoft.com/query/dev14.query?appId=Dev14IDEF1&l=EN-US&k=k:System.Int64 'System.Int64') | The entity identity value. |
+| ignoreQueryFilters | [System.Boolean](http://msdn.microsoft.com/query/dev14.query?appId=Dev14IDEF1&l=EN-US&k=k:System.Boolean 'System.Boolean') | If true, global query filters will be ignored for this query. |
+| includeProperties | [System.Linq.Expressions.Expression{System.Func{\`0,System.Object}}[]](http://msdn.microsoft.com/query/dev14.query?appId=Dev14IDEF1&l=EN-US&k=k:System.Linq.Expressions.Expression 'System.Linq.Expressions.Expression{System.Func{`0,System.Object}}[]') | Navigation properties that should be included. |
+
+<a name='M-AndcultureCode-CSharp-Core-Interfaces-Conductors-IRepositoryReadConductor`1-FindById-System-Int64,System-String[]-'></a>
+### FindById(id,includeProperties) `method`
```
Comment (jstevenson) [inline src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md:2991]: And here

File: src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md [hunk 4]
```diff
@@ -3104,6 +3335,149 @@ true
...
+##### Parameters
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| id | [System.Int64](http://msdn.microsoft.com/query/dev14.query?appId=Dev14IDEF1&l=EN-US&k=k:System.Int64 'System.Int64') | The entity identity value. |
+| ignoreQueryFilters | [System.Nullable{System.Boolean}](http://msdn.microsoft.com/query/dev14.query?appId=Dev14IDEF1&l=EN-US&k=k:System.Nullable 'System.Nullable{System.Boolean}') | If true, global query filters will be ignored for this query. |
+| includeProperties | [System.Linq.Expressions.Expression{System.Func{\`0,System.Object}}[]](http://msdn.microsoft.com/query/dev14.query?appId=Dev14IDEF1&l=EN-US&k=k:System.Linq.Expressions.Expression 'System.Linq.Expressions.Expression{System.Func{`0,System.Object}}[]') | Navigation properties that should be included. |
+
+<a name='M-AndcultureCode-CSharp-Core-Interfaces-Data-IRepository`1-FindById-System-Int64,System-String[]-'></a>
+### FindById(id,includeProperties) `method`
```
Comment (jstevenson) [inline src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md:3412]: And here

**Review by brandongregoryscott — APPROVED** [review]

**Comment by brandongregoryscott** [comment]
Thanks for taking the extra time to fill in the XML comments, too!

**Review by jstevenson — CHANGES_REQUESTED** [review]

File: src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md (hunk 1 above)
Comment (brandongregoryscott) [reply src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md:214]: This file is autogenerated 👍

Comment (jstevenson) [reply src/AndcultureCode.CSharp.Core/AndcultureCode.CSharp.Core.md:214]: > This file is autogenerated 👍

Said just mentioned that to me also. I guess the source was out of order.  🤦

**Comment by wintondeshong** [comment]
@all-contributors please add @SaidShah for documentation

**Pull request merged**